examples/engine/1_basics.py

```diff
@@ -1,46 +1,64 @@
+# %% [markdown]
 """
-1. Basics
-=========
+# 1. Basics
+
+This notebook shows basic example of creating a simple dialog bot (agent).
+Let's do all the necessary imports from `dff`:
 """
 
-# TODO:
-# 1. The illustration should express only what is shown in the example, not all at once.
-#    Complicate illustrations from one example to another.
-# 2. Fix PRE_RESPONSE_PROCESSING to PRE_TRANSITIONS_PROCESSING
-# 3. Specify that the nodes form an ordered set
 
+# %%
+# pip install dff  # Uncomment this line to install the framework
+
+
+# %%
+from dff.core.engine.core import Actor
+from dff.core.pipeline import Pipeline
 from dff.core.engine.core.keywords import TRANSITIONS, RESPONSE
 import dff.core.engine.conditions as cnd
 
-from dff.core.pipeline import Pipeline
-from dff.utils.testing.common import check_happy_path, is_interactive_mode, run_interactive_mode
+from dff.utils.testing.common import (
+    check_happy_path,
+    is_interactive_mode,
+    run_interactive_mode,
+)
+
 
-# First of all, to create a dialog agent, we need to create a dialog script.
-# Below, `script` is the dialog script.
-# A dialog script is a flow dictionary that can contain multiple script .
-# Script are needed in order to divide a dialog into sub-dialogs and process them separately.
-# For example, the separation can be tied to the topic of the dialog.
-# In our example, there is one flow called greeting_flow.
+# %% [markdown]
+"""
+First of all, to create a dialog agent, we need to create a dialog script.
+Below script means a dialog script.
+A script is a dictionary, where the keys are the names of the flows.
+A script can contain multiple scripts, what is needed in order to divide
+a dialog into sub-dialogs and process them separately.
+For example, the separation can be tied to the topic of the dialog.
+In this example there is one flow called `greeting_flow`.
+
+Flow describes a sub-dialog using linked nodes.
+Each node has the keywords `RESPONSE` and `TRANSITIONS`.
 
-# Inside each flow, we can describe a sub-dialog.
-# Here we can also use keyword `LOCAL`, which we have considered in other examples.
+* `RESPONSE` contains the response
+    that the agent will return from the current node.
+* `TRANSITIONS` describes transitions from the
+current node to another nodes. This is a dictionary,
+where keys are names of the nodes and
+values are conditions of transition to them.
```

Comment:
Shouldn't there be an indentation?
```suggestion
* `TRANSITIONS` describes transitions from the
    current node to another nodes. This is a dictionary,
    where keys are names of the nodes and
    values are conditions of transition to them.
```

Reply:
This does not affect the final text presentation, but can be done to improve the readability of the code.